ci/docker-db.sh

```diff
@@ -1,59 +1,76 @@
+export MYSQL_HOST=${MYSQL_HOST:-127.0.0.1}
 export MYSQL_TCP_PORT=${MYSQL_TCP_PORT:-13306}
-export PGHOST=127.0.0.1
-NAME="hub-test-$DB"
-DOCKER_RUN="docker run -d --name $NAME"
-
-docker rm -f "$NAME" 2>/dev/null || true
-
-case "$DB" in
-"mysql")
-  RUN_ARGS="-e MYSQL_ALLOW_EMPTY_PASSWORD=1 -p $MYSQL_TCP_PORT:3306 mysql:5.7"
-  CHECK="mysql --host $MYSQL_HOST --port $MYSQL_TCP_PORT --user root -e \q"
-  ;;
-"postgres")
-  RUN_ARGS="-p 5432:5432 postgres:9.5"
-  CHECK="psql --user postgres -c \q"
-  ;;
-*)
-  echo '$DB must be mysql or postgres'
-  exit 1
-esac
-
-$DOCKER_RUN $RUN_ARGS
+export PGHOST=${PGHOST:-127.0.0.1}
+export PGPORT=${PGPORT:-5432}
 
+# Define a name for a database container to start
+DOCKER_CONTAINER="hub-test-$DB"
+
+# Stop and remove any existing database container
+docker rm -f "$DOCKER_CONTAINER" 2>/dev/null || true
+
+# Prepare env vars DOCKER_RUN_ARGS and READINESS_CHECK
+if [[ "$DB" == "mysql" ]]; then
+    # Environment variables are considered by the Docker image, and the mysql
+    # client respectively.
+    # ref, docker: https://hub.docker.com/_/mysql/
+    # ref, client: https://dev.mysql.com/doc/refman/5.7/en/setting-environment-variables.html
+    #
+    DOCKER_RUN_ARGS="-p $MYSQL_TCP_PORT:3306 --env MYSQL_ALLOW_EMPTY_PASSWORD=1 mysql:5.7"
+    READINESS_CHECK="mysql --user root --execute \q"
+elif [[ "$DB" == "postgres" ]]; then
+    # Environment variables are considered by the Docker image, and the psql
+    # client respectively.
+    # ref, docker: https://hub.docker.com/_/postgres/
+    # ref, client: https://www.postgresql.org/docs/9.5/libpq-envars.html
+    #
+    DOCKER_RUN_ARGS="-p $PGPORT:5432 postgres:9.5"
+    READINESS_CHECK="psql --user postgres --command \q"
+else
+    echo '$DB must be mysql or postgres'
+    exit 1
+fi
+
+# Start a new container
+docker run --detach --name "$DOCKER_CONTAINER" $DOCKER_RUN_ARGS
+
+# Wait for the new container to start
 echo -n "waiting for $DB "
 for i in {1..60}; do
-  if $CHECK; then
-    echo 'done'
-    break
-  else
-    echo -n '.'
-    sleep 1
-  fi
+    if $READINESS_CHECK; then
+        echo 'done'
+        break
+    else
+        echo -n '.'
+        sleep 1
+    fi
 done
-$CHECK
-
-case "$DB" in
-"mysql")
-  ;;
-"postgres")
-  # create the user
-  psql --user postgres -c "CREATE USER $PGUSER WITH PASSWORD '$PGPASSWORD';"
-  ;;
-*)
-esac
-
-echo -e "
-Set these environment variables:
+$READINESS_CHECK
 
-    export MYSQL_HOST=127.0.0.1
+# Setup additional users to mysql's root and postgresql's postgres users
+if [[ "$DB" == "mysql" ]]; then
+    echo "A dedicated user for mysql isn't created."
+elif [[ "$DB" == "postgres" ]]; then
+    psql --user postgres --command "CREATE USER $PGUSER WITH PASSWORD '$PGPASSWORD';"
+fi
+
+# Print message for humans running the script
+if [[ "$DB" == "mysql" ]]; then
+    echo -e "
+Set these environment variables:
+    export MYSQL_HOST=$MYSQL_HOST
```

Comment:
as above, echoing HOST doesn't provide useful info, since it's either

1. already set and doesn't need to be set again, or
2. not set, and doesn't need to be set because it's the default